core/src/main/scala/caliban/GraphQL.scala

```diff
@@ -58,12 +58,10 @@ trait GraphQL[-R] { self =>
    * adding some middleware around the query execution.
    * Fails with a [[caliban.CalibanError.ValidationError]] if the schema is invalid.
    */
-  final def interpreter(implicit trace: Trace): IO[ValidationError, GraphQLInterpreter[R, CalibanError]] = {
-    val i = cachedInterpreter
-    ZIO.fromEither(i)
-  }
+  final def interpreter(implicit trace: Trace): IO[ValidationError, GraphQLInterpreter[R, CalibanError]] =
+    ZIO.fromEither(interpreterEither)
 
-  private lazy val cachedInterpreter =
+  private[caliban] final lazy val interpreterEither =
```

Comment:
PS: I'm actually thinking we should change the `interpreter` method to return an Either instead. I think it can be quite annoying to users to have to deal with the ZIO effect unnecessarily

Reply:
Isn't an `Either` more annoying? With ZIO you can just call it in your current for-comprehension, with `Either` I feel it will make people have to call `ZIO.fromEither`.

Reply:
I think it'll simplify things a lot for non-zio apps. I think this is the only method required for wiring the api that returns a ZIO effect